src/bind.c

```diff
@@ -75,9 +75,7 @@ static SEXP vec_rbind(SEXP xs, SEXP ptype) {
     }
 
     SEXP tbl = PROTECT(vec_cast(VECTOR_ELT(xs, i), ptype));
-    idx_ptr[0] = counter;
-    idx_ptr[1] = counter + size;
-
+    init_compact_seq(idx_ptr, counter, counter + size);
```

Comment:
I mean get rid of `idx_ptr` altogether and just pass two integers. I think this change actually makes the code harder to read because it's not immediately obvious what the connection between this line and the next is. 

Wouldn't `df_assign(out, counter, counter + size, tbl, false)` be more clear?

Reply:
It can't be `df_assign()`, because this is the code that takes an R index. We could have a variant with the signature you suggest but then that variant would have to allocate a compact idx at each iteration, which can be many when assembling vectors with `!!!`. Alternatively we could duplicate all the slice/assign functions, but I think this wouldn't be desirable.